File: changsukuaidi-express-andriod/app/src/main/java/com/changsukuaidi/www/modules/login/LoginFragment.java
package com.changsukuaidi.www.modules.login;

import android.content.Intent;
import android.text.TextUtils;

import com.changsukuaidi.www.R;
import com.changsukuaidi.www.base.BaseApplication;
import com.changsukuaidi.www.base.BaseFragment;
import com.changsukuaidi.www.bean.CarBean;
import com.changsukuaidi.www.databinding.FragmentLoginBinding;
import com.changsukuaidi.www.modules.home.HomeActivity;
import com.changsukuaidi.www.modules.login.forget_password.ForgetPwdActivity;
import com.changsukuaidi.www.modules.punch.PunchActivity;
import com.jakewharton.rxbinding2.view.RxView;
import com.jakewharton.rxbinding2.widget.RxTextView;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;

/**
 * @Describe
 * @Author thc
 * @Date 2017/12/6
 */

public class LoginFragment extends BaseFragment<LoginPresenter, FragmentLoginBinding> {

    @Override
    protected void setPresenter() {
        DaggerLoginComp.builder()
                .appComp(BaseApplication.getAppComp())
                .loginModule(new LoginModule(this))
                .build()
                .injectMembers(this);
    }

    @Override
    protected int layoutId() {
        return R.layout.fragment_login;
    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initView() {
        // 设置顶部Toolbar
        setToolbarTitle(getString(R.string.login));

        // 监听输入框输入
        Observable.combineLatest(RxTextView.textChanges(mViewBinding.phoneText), RxTextView.textChanges(mViewBinding.passwordText),
                (charSequence, charSequence2) -> !TextUtils.isEmpty(charSequence) && !TextUtils.isEmpty(charSequence2))
                .subscribe(aBoolean -> mViewBinding.loginBtn.setEnabled(aBoolean));

        RxView.clicks(mViewBinding.loginBtn)
                .throttleFirst(1, TimeUnit.SECONDS)
                .compose(this.bindToLifecycle())
                .subscribe(aVoid ->
                        // 密码登录
                        mPresenter.login(mViewBinding.phoneText.getText().toString(), mViewBinding.passwordText.getText().toString()));

        RxView.clicks(mViewBinding.forgetPwdText)
                .compose(this.bindToLifecycle())
                .subscribe(aVoid ->
                        // 忘记密码
                        startActivity(new Intent(getActivity(), ForgetPwdActivity.class)));

    }

    /**
     * 登录成功去获取打卡和车辆信息
     */
    public void loginSuccess() {
        mPresenter.getCarInfo();
    }

    /**
     * 判断是否是打卡，进入打卡页面获取主页
     *
     * @param carBean
     */
    public void finishGetCarInfo(CarBean carBean) {
        if (carBean.getPunchInfo() == null || TextUtils.isEmpty(carBean.getPunchInfo().getStime())) {
            startActivity(new Intent(getActivity(), PunchActivity.class));
        } else {
            startActivity(new Intent(getActivity(), HomeActivity.class));
        }
        getActivity().overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        getActivity().finish();
    }
}

File: WIFIDirect/app/src/main/java/com/thc/www/p2p/android/login/LoginFragment.kt
package com.thc.www.p2p.android.login

import com.jakewharton.rxbinding2.widget.RxTextView
import com.jakewharton.rxbinding2.widget.TextViewAfterTextChangeEvent
import com.thc.www.p2p.base.BaseApplication
import com.thc.www.p2p.base.BaseFragment
import com.thc.www.p2p.comp.DaggerLoginComp
import com.thc.www.p2p.module.LoginModule
import com.thc.www.wifi_direct.R
import io.reactivex.Observable
import io.reactivex.functions.BiFunction
import kotlinx.android.synthetic.main.fragment_login.*

/**
 * @Describe
 * @Author thc
 * @Date  2018/5/7
 */
class LoginFragment : BaseFragment<LoginPresenter>() {
    override fun setPresenter() {
        DaggerLoginComp.builder()
                .appComp((activity!!.application as BaseApplication).getAppComp())
                .loginModule(LoginModule(this))
                .build()
                .injectMembers(this)
    }

    override fun getLayoutId(): Int = R.layout.fragment_login

    override fun initView() {
        setToolbarTitle("登录")

        Observable.combineLatest(RxTextView.afterTextChangeEvents(det_login_account),
                RxTextView.afterTextChangeEvents(det_login_pwd),
                BiFunction<TextViewAfterTextChangeEvent, TextViewAfterTextChangeEvent, Boolean> { s1, s2 ->
                    s1.editable()!!.length >= 6 && s2.editable()!!.length >= 6
                }).subscribe({
            btn_login.isEnabled = it
        })

        btn_login.setOnClickListener {
            mPresenter?.login(det_login_account.text.toString(), det_login_pwd.text.toString())
        }
    }

    override fun useToolBar(): Boolean = true
}